data-serving/data-service/src/controllers/case.ts

```diff
@@ -550,25 +561,32 @@ export class CasesController {
                     c.caseReference?.sourceId &&
                     c.caseReference?.sourceEntryId
                 ) {
-                    return {
-                        updateOne: {
-                            filter: {
-                                'caseReference.sourceId':
-                                    c.caseReference.sourceId,
-                                'caseReference.sourceEntryId':
-                                    c.caseReference.sourceEntryId,
+                    const day0case = await Day0Case.findOne({
+                        'caseReference.sourceId': c.caseReference.sourceId,
+                        'caseReference.sourceEntryId':
+                            c.caseReference.sourceEntryId,
+                    });
+                    if (day0case) {
+                        return {
+                            updateOne: {
+                                filter: {
+                                    'caseReference.sourceId':
+                                        c.caseReference.sourceId,
+                                    'caseReference.sourceEntryId':
+                                        c.caseReference.sourceEntryId,
+                                },
+                                update: c,
```

Comment:
This is the part that I had the most problems with.
Identifier is being set in the middleware during `save` operation. However the `bulkWrite` method does not trigger this middleware. So I had to somehow work around that. I had two options:
1) Check if the cases that are ment to be upsert already are in database. If so only update them.
2) Ditch the `bulkWrite` completely and perform `save` for each of the resources in the upsert.

I decided for the first option, but I'm not confident that it is better.

I'd like to get your opinion on that. Or maybe you have some other solution?

Reply:
Yes, but previously this `updateOne` function had parameter `upsert: true` so when the data for case was sent and filter parameters did not match any case already existing case was being created. So we did not know whether the case is new or not, thats why I needed to check first, and then use `setId`